### DISCUSSION TITLE
50 million + products in Pimcore, MySQL

### DISCUSSION THREAD

**Discussion by al3wb** [pimcore/pimcore#14912] [Q&A / Support]
Hello!

We are creating a store of affiliate products. Approximately 1800 categories (20 large and many nested subsidiaries). About 50 million products at the start (in the future will be more).

Will Pimcore and MySQL be able to handle such a number of products? Can Pimcore work with a sharded MySQL database?

We have a lot of servers. Most likely for such a number of products need to split Pimcore and the database, but how to do it is not clear.

I don't want to write a self-written system from scratch, I would like to use the ready functionality of Pimcore.

**Comment by al3wb** [comment]
Hello!
Is no one interested in the issue of such large-scale projects on Pimcore? =)

Or maybe there's documentation on the Pimcore website about this kind of scaling and I just haven't noticed it?

**Comment by pdchaudhary** [comment]


**Reply by al3wb** [reply]
Maybe there are some descriptions of approaches to implement Pimcore with 50 - 100 million products? Maybe in paid courses on Pimcore?

After all, you will most likely need to split MySQL database by shards. It is unclear how to configure Pimcore.

**Reply by dpfaffenbauer** [reply]
Pimcore doesn't support shards. 50-100 million is honestly a challenge for Pimcore. We have a customer with 2 million and creates all sorts of challenges. 

There is no answer here cause there isn't really one.
